packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -171,7 +205,11 @@ impl Resolver {
     ///
     /// Arguments:
     /// - `flat_fragment`: a `FlatFragment` containing the new descendants added to `flat_fragment.parent_idx`.
-    pub fn add_nodes(&mut self, flat_fragment: &FlatFragment) {
+    pub fn add_nodes(
+        &mut self,
+        flat_fragment: &FlatFragment,
+        as_index_resolutions: Option<AsIndexResolutions>,
```

Comment:
Not a fan of the `Option` here. Maybe add a `ResolutionType::Node(...)` and `ResolutionType::Index(...)` enum to wrap `flat_fragment`.